targets/wasm_exec.js

```diff
@@ -319,6 +319,25 @@
 					//"syscall/js.valueInstanceOf": (sp) => {
 					//	mem().setUint8(sp + 24, loadValue(sp + 8) instanceof loadValue(sp + 16));
 					//},
+
+					// copyBytesToJS(dst ref, src []byte) (int, bool)
+					// Originally copied from upstream Go project, then modified:
+					//   https://github.com/golang/go/blob/3f995c3f3b43033013013e6c7ccc93a9b1411ca9/misc/wasm/wasm_exec.js#L404-L416
+					"syscall/js.copyBytesToJS": (ret_addr, dest_addr, source_addr, source_len, source_cap) => {
+						let num_bytes_copied_addr = ret_addr;
+						let returned_status_addr = ret_addr + 4; // Address of returned boolean status variable
+
+						const dst = loadValue(dest_addr);
+						const src = loadSlice(source_addr);
```

Comment:
I think you need to pass the `len` as well (`source_len`).

Looking at `loadSlice` it seems to ignore the `cap` parameter.

Reply:
Not sure.  Looking at the upstream Go version it's copied from, that one doesn't seem to use it?

https://github.com/golang/go/blob/3f995c3f3b43033013013e6c7ccc93a9b1411ca9/misc/wasm/wasm_exec.js#L404-L416

Reply:
Ahhh.  The TinyGo `loadslice()` defined earlier in the file.

Will update. :smile:

Reply:
Done.

Reply:
Thank you!